lib/carrier_wave/storage/git_hub.rb

```diff
@@ -0,0 +1,94 @@
+module CarrierWave
+  module Storage
+    class GitHub < File
+
+      def store!(file)
+        upload_path = "#{SecureRandom.uuid}/#{file.original_filename}"
```

Comment:
We should partition the root tree so it doesn't get huge. I'm inclined to copy how Git stores its object database:

```ruby
blob = repo.git.blobs.create(...)
uploadPath = "#{blob[0,2]}/#{blob[2,38]}/#{file.original_filename}"
```